tianshou/data/utils.py

```diff
@@ -24,7 +24,9 @@ def to_torch(x: Union[torch.Tensor, dict, Batch, np.ndarray],
              device: Union[str, int, torch.device] = 'cpu'
              ) -> Union[dict, Batch, torch.Tensor]:
     """Return an object without np.ndarray."""
-    if isinstance(x, np.ndarray):
+    if isinstance(x, (np.generic, np.ndarray)):
+        if isinstance(x, np.generic):
+            x = np.asanyarray(x)
```

Comment:
still raise an exception:
```python
In [17]: a=np.array([float('nan'), np.float32])[0]

In [18]: a
Out[18]: nan

In [19]: to_torch(a).dtype
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-19-156efb624c56> in <module>
----> 1 to_torch(a).dtype

AttributeError: 'float' object has no attribute 'dtype'
```

Reply:
Do you think it should not in this case?

Reply:
Emm...I think it may be a case in Batch.to_torch, but it works well:
```python
In [46]: a=Batch(a=np.array([0., float('nan')], np.float32))

In [47]: to_torch(a[1])
Out[47]: 
Batch(
    a: tensor(nan),
)
```

Reply:
No it's because you are confusing 2 different things: you are comparing `np.array([float('nan'), np.float32])`, which is equal to `array([nan, <class 'numpy.float32'>], dtype=object)`, since `np.float32` is a class, not an instance. And when you access an item from a numpy array of object, it completely remove numpy wrapping, so `a[0]` is not longer an numpy scalar in the first place, but a native python scalar.

Reply:
The point is: does `to_torch([0., float('nan')])` should return a tensor ? For `Batch` it is not an issue, since the list is automatically converted in `np.ndarray` and therefore converted in `Tensor`.

Reply:
No, I'm not packing nan with np.float32, instead `np.array([nan], np.float32)`

Reply:
> The point is: does to_torch([0., float('nan')]) should return a tensor ? For Batch.to_torch it is not an issue, since the list is automatically converted in np.ndarray and therefore converted in Tensor.

I think it should.

Reply:
Oh sorry, I just notice that my first case is wrong :(